File: net/OpenDraft.Lang/StringLiteralToken.cs
// <copyright file="StringLiteralToken.cs" company="Jon Rowlett">
// Copyright (c) Jon Rowlett. All rights reserved.
// </copyright>

namespace OpenDraft.Lang;

/// <summary>
/// Represents a string literal token in the language.
/// </summary>
public class StringLiteralToken : Token
{
    /// <summary>
    /// Initializes a new instance of the <see cref="StringLiteralToken"/> class.
    /// </summary>
    /// <param name="source">The source reference for this token.</param>
    /// <param name="value">The value of the string literal.</param>
    /// <exception cref="ArgumentNullException">Thrown if <paramref name="source"/> or <paramref name="value"/> is null.</exception>
    public StringLiteralToken(SourceReference source, string value)
        : base(source)
    {
        this.Value = value ?? throw new ArgumentNullException(nameof(value));
    }

    /// <summary>
    /// Gets the value of the string literal.
    /// </summary>
    public string Value { get; }
}

File: java/src/io/github/localtoast9001/opendraft/lang/StringLiteralToken.java
/**
 * Copyright (c) Jon Rowlett. All rights reserved.
 */
package io.github.localtoast9001.opendraft.lang;

/**
 * Represents a string literal token in the OpenDraft language.
 */
public final class StringLiteralToken extends Token {
    private final String value;

    /**
     * Creates a new instance of the StringLiteralToken class.
     * @param source The source reference of the token.
     * @param value The string value of the token.
     * @throws IllegalArgumentException if value is null.
     */
    public StringLiteralToken(SourceReference source, String value) {
        super(source);
        if (value == null) {
            throw new IllegalArgumentException("Value cannot be null");
        }
        this.value = value;
    }

    /**
     * Returns the string value of this token.
     * @return the string value
     */
    public String getValue() {
        return value;
    }

    @Override
    public String toString() {
        return "StringLiteralToken{"
            + "value='" + value + '\''
            + '}';
    }
}
